### PR TITLE
doc: enable Graphviz extension

### PR DESCRIPTION
**Pull request by gmarull**
```
doc: enable graphviz extension

Enable the built-in Sphinx Graphviz extension to allow creating Graphviz
diagrams natively on the source files. Some style defaults have been
enabled to make sure diagrams are consistent and have good quality in
both light and dark modes.
```
```
doc: guides: docs: replace diagram image with dot source …

Replace the dot diagram with its original source. This is possible
thanks to the Sphinx Graphviz extension. Note that some style attributes
have been removed as defaults are already provided by the extension.
```

Before:
image

After:
image

### CODE REVIEW

**Review by nashif — APPROVED** [review]

File: doc/guides/docs/index.rst [hunk 1]
```diff
@@ -55,6 +55,7 @@ Installing the documentation processors
 Our documentation processing has been tested to run with:
 
 * Doxygen version 1.8.13
+* Graphviz 2.43
```
Comment (mbolivar-nordic) [inline doc/guides/docs/index.rst:80]: This is good, but I think you forgot to update the `apt-get` etc. lines below.

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]

File: doc/guides/docs/index.rst (hunk 1 above)
Comment (gmarull) [reply doc/guides/docs/index.rst:80]: good catch, fixed

**Review by mbolivar-nordic — COMMENTED** [review]
Thanks for the update. Did you test on macOS and Windows?

**Comment by gmarull** [comment]
> Thanks for the update. Did you test on macOS and Windows?

No, I haven't tested, but I assume that if `dot` is found, docs will build as expected (CMake is helping here with `find_package(Doxygen REQUIRED dot)`).

**Comment by mbolivar-nordic** [comment]
> No, I haven't tested,

OK, I will leave my -1 in place for now then. I can test for you later today when I have access to my windows and macOS boxes and will approve if all goes well.

**Comment by ceolin** [comment]
Oo that is gold. Does it work with PDF generation as well ? The image generated and then included in the generated latex ?

**Comment by gmarull** [comment]
> Oo that is gold. Does it work with PDF generation as well ? The image generated and then included in the generated latex ?

Yes, PDF build works correctly (I tested it).

**Comment by mbolivar-nordic** [comment]
Mac / dark:

Screen Shot 2021-07-20 at 4 45 44 PM

**Comment by mbolivar-nordic** [comment]
Windows / light:

light

**Review by mbolivar-nordic — APPROVED** [review]
Looks good after spot checking on windows and mac. Thanks; this will make it much easier to maintain these.

**Pull request merged**